src/PlottingInterface.cc

```diff
@@ -0,0 +1,466 @@
+    return ref->GetFloat(_msg, _field);
+  else if (type == FieldDescriptor::Type::TYPE_INT32)
+    return ref->GetInt32(_msg, _field);
+  else if (type == FieldDescriptor::Type::TYPE_INT64)
+    return ref->GetInt64(_msg, _field);
+  else if (type == FieldDescriptor::Type::TYPE_BOOL)
+    return ref->GetBool(_msg, _field);
+  else if (type == FieldDescriptor::Type::TYPE_UINT32)
+    return ref->GetUInt32(_msg, _field);
+  else if (type == FieldDescriptor::Type::TYPE_UINT64)
+    return ref->GetUInt64(_msg, _field);
+  else
+  {
+    ignwarn << "Non Plotting Type" << std::endl;
+    return 0;
+  }
+}
+
+////////////////////////////////////////////
+Transport::Transport() : dataPtr(std::make_unique<TransportPrivate>())
+{
+}
+
+////////////////////////////////////////////
+Transport::~Transport()
+{
+  // unsubscribe from all topics in the transport
+  for (auto topicIt = this->dataPtr->topics.begin();
+    topicIt != this->dataPtr->topics.end(); ++topicIt)
+    this->dataPtr->node.Unsubscribe(topicIt->first);
+}
+
+////////////////////////////////////////////
+void Transport::Unsubscribe(const std::string &_topic,
+                            const std::string &_fieldPath,
+                            int _chart)
+{
+  if (this->dataPtr->topics.count(_topic))
+  {
+    this->dataPtr->topics[_topic]->UnRegister(_fieldPath, _chart);
+
+    // if there is no registered fields, unsubscribe from the topic
+    if (this->dataPtr->topics[_topic]->FieldCount() == 0)
+    {
+      this->dataPtr->node.Unsubscribe(_topic);
+      this->dataPtr->topics.erase(_topic);
+    }
+  }
+}
+
+////////////////////////////////////////////
+void Transport::Subscribe(const std::string &_topic,
+                          const std::string &_fieldPath,
+                          int _chart)
+{
+  // new topic
+  if (this->dataPtr->topics.count(_topic) == 0)
+  {
+    auto topicHandler = new Topic(_topic);
+    this->dataPtr->topics[_topic] = topicHandler;
+
+    topicHandler->Register(_fieldPath, _chart);
+    this->dataPtr->node.Subscribe(_topic, &Topic::Callback, topicHandler);
+  }
+  // already exist topic
+  else
+  {
+    this->dataPtr->topics[_topic]->Register(_fieldPath, _chart);
+    this->dataPtr->node.Subscribe(_topic, &Topic::Callback,
+                                  this->dataPtr->topics[_topic]);
+  }
+}
+
+//////////////////////////////////////////////////////
+const std::map<std::string, Topic*> &Transport::Topics()
+{
+  return this->dataPtr->topics;
+}
+
+//////////////////////////////////////////////////////
+bool Transport::TopicFound(const std::string &_topic) const
+{
+  // check if the topic exist
+  std::vector<std::string> topics;
+  this->dataPtr->node.TopicList(topics);
+  auto foundTopic = std::find(topics.begin(), topics.end(), _topic);
+  // topic is unsubscribed ... update the model
+  if (foundTopic == topics.end())
+  {
+    // TODO(Amr) : unsubscribe from the topic and remove its UI fields
```

Comment:
This sounds like something strange to do inside this function. As a user, I'd expect just to receive `true` / `false`, without any other side effects.